plugins/trendmicro_visionone/icon_trendmicro_visionone/triggers/poll_oat_list/trigger.py

```diff
@@ -0,0 +1,95 @@
+        # Get Connection Client
+        client = self.connection.client
+        # Get Action Parameters
+        detected_start_date_time = params.get(Input.DETECTED_START_DATE_TIME)
+        detected_end_date_time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")  # Current time in UTC
+        ingested_start_date_time = params.get(Input.INGESTED_START_DATE_TIME)
+        ingested_end_date_time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")  # Current time in UTC
+        fields = params.get(Input.FIELDS)
+        query_op = params.get(Input.QUERY_OP)
+        # Choose enum
+        if "or" in query_op:
+            query_op = pytmv1.QueryOp.OR
+        elif "and" in query_op:
+            query_op = pytmv1.QueryOp.AND
+
+        while True:
+            # Make sure response contains < 5K entries
+            self.logger.info("Checking Response Size...")
+            count = client.oat.list(
+                detected_start_date_time=detected_start_date_time,
+                detected_end_date_time=detected_end_date_time,
+                ingested_start_date_time=ingested_start_date_time,
+                ingested_end_date_time=ingested_end_date_time,
+                top=1,
+                op=query_op,
+                **fields,
+            )
+            if "error" in count.result_code.lower():
+                raise PluginException(
+                    cause="An error occurred while getting OAT count.",
+                    assistance="Please check your inputs and try again.",
+                    data=count.error,
+                )
+            total_count = count.response.total_count
+            if total_count >= 50:
+                raise PluginException(
+                    cause="Attempted query is over-sized (more than 50 results).",
+                    assistance="Please refine your inputs to reduce search size and try again.",
+                )
+            new_oats = []
+            # Get OATs List
+            self.logger.info("Getting OATs List...")
+            try:
+                client.oat.consume(
+                    lambda oat: new_oats.append(oat.model_dump_json()),
+                    detected_start_date_time=detected_start_date_time,
+                    detected_end_date_time=detected_end_date_time,
+                    ingested_start_date_time=ingested_start_date_time,
+                    ingested_end_date_time=ingested_end_date_time,
+                    top=50,
+                    op=query_op,
+                    **fields,
+                )
+            except Exception as error:
+                raise PluginException(
+                    cause="An error occurred while polling OATs.",
+                    assistance="Please check your inputs and try again.",
+                    data=error,
+                )
+            # Load json objects to list
+            oats_list = []
+            for new_oat in new_oats:
+                oats_list.append(json.loads(new_oat))
```

Comment:
Same here. I think its unnecessary, also you are able to decode that to JSON at the consume level above